node/src/components/chainspec_loader/chainspec.rs

```diff
@@ -231,6 +231,12 @@ pub struct GenesisConfig {
     pub(crate) wasm_config: WasmConfig,
     pub(crate) deploy_config: DeployConfig,
     pub(crate) highway_config: HighwayConfig,
+    /// Initial era ID for cases when we restart the network after a failure and want to start from
+    /// an era other than 0.
+    pub(crate) initial_era_id: Option<EraId>,
+    /// Initial block height for cases when we restart the network after a failure and want to
+    /// start from a block height other than 0.
+    pub(crate) initial_block_height: Option<u64>,
```

Comment:
maybe you answer this somewhere, but what happens if someone configures an initial_block_height  and an initial_era_id and the specified block is an era other than initial_era_id?

Reply:
This just defines which block height we will assign to the next block we create, so that shouldn't really be possible. Unless you mean: what if we, at some point, download a conflicting block at the same height with another era ID - then storage will inform us that we are attempting to put a duplicate block at the same height.
Which, actually, looks like something that could happen in practice after a restart. We could either restart from a block height that was not the latest block in the chain, or we could have a situation in which some nodes finalized more blocks than others, and already have conflicting blocks in storage. In this case these nodes wouldn't be able to store the blocks finalized after the restart :grimacing: I'm not sure what the best solution would be here.